Signal/src/environment/ExperienceUpgradeFinder.swift

```diff
@@ -6,27 +6,45 @@ import Foundation
+                                 image: #imageLiteral(resourceName: "introductory_splash_video_calling"))
+    }
+
+    var callKit: ExperienceUpgrade {
+        return ExperienceUpgrade(uniqueId: ExperienceUpgradeId.callKit.rawValue,
+                                 title: NSLocalizedString("UPGRADE_EXPERIENCE_CALLKIT_TITLE", comment: "Header for upgrade experience"),
+                                 body: NSLocalizedString("UPGRADE_EXPERIENCE_CALLKIT_DESCRIPTION", comment: "Description of CallKit to upgrading (existing) users"),
+                                 image: #imageLiteral(resourceName: "introductory_splash_callkit"))
+    }
+
+    var introducingProfiles: ExperienceUpgrade {
+        return ExperienceUpgrade(uniqueId: ExperienceUpgradeId.introducingProfiles.rawValue,
+                                 title: NSLocalizedString("UPGRADE_EXPERIENCE_INTRODUCING_PROFILES_TITLE", comment: "Header for upgrade experience"),
+                                 body: NSLocalizedString("UPGRADE_EXPERIENCE_INTRODUCING_PROFILES_DESCRIPTION", comment: "Description of new profile feature for upgrading (existing) users"),
+                                 image:#imageLiteral(resourceName: "introductory_splash_profile"))
+    }
+
     // Keep these ordered by increasing uniqueId.
     private var allExperienceUpgrades: [ExperienceUpgrade] {
-        var upgrades = [ExperienceUpgrade(uniqueId: ExperienceUpgradeId.videoCalling.rawValue,
-                                          title: NSLocalizedString("UPGRADE_EXPERIENCE_VIDEO_TITLE", comment: "Header for upgrade experience"),
-                                          body: NSLocalizedString("UPGRADE_EXPERIENCE_VIDEO_DESCRIPTION", comment: "Description of video calling to upgrading (existing) users"),
-                                          image: #imageLiteral(resourceName: "introductory_splash_video_calling"))]
-
-        if UIDevice.current.supportsCallKit {
-            upgrades.append(ExperienceUpgrade(uniqueId: ExperienceUpgradeId.callKit.rawValue,
-                                              title: NSLocalizedString("UPGRADE_EXPERIENCE_CALLKIT_TITLE", comment: "Header for upgrade experience"),
-                                              body: NSLocalizedString("UPGRADE_EXPERIENCE_CALLKIT_DESCRIPTION", comment: "Description of CallKit to upgrading (existing) users"),
-                                              image: #imageLiteral(resourceName: "introductory_splash_callkit")))
-        }
-
-        return upgrades
+        return [
+            // Disable old experience upgrades. Most people have seen them by now, and accomodating multiple makes layout harder.
+            // Note if we ever want to show multiple experience upgrades again
+            // we'll have to update the layout in ExperienceUpgradesPageViewController
+            //
+            // videoCalling,
+            // (UIDevice.current.supportsCallKit ? callKit : nil),
```

Comment:
`ExperienceUpgradesPageViewController` still has paging... is that for future scenarios where we might want to have two pages?

Reply:
Correct.